### PR TITLE
Added Components

### PR DESCRIPTION
**Pull request by Alqanar**
Added components that use only in the Blog:
- ArticleCard, 
- BlogImage, 
- Tags,
- Pagination

Added common components with repository next:
- Box, 
- Flex, 
- Button, 
- EmailSubscribe, 
- as well as the MarketoForm component in the part used by the EmailSubscribe component,
- Icon, 
- Image,
- Link,
- Section

### CODE REVIEW

File: components/Link/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  if (isHash(href) || isExternalLink(href) || isLocalAssetFile(href)) {
+    return href;
+  }
+
+  const currentHref = normalizePath(asPath);
+  let fullHref = resolve(splitPath(currentHref).path, href);
+
+  if (fullHref.startsWith("/docs/")) {
+    fullHref = updateScopeInUrl(fullHref, scope);
+  }
```
Comment (iAdramelk) [inline components/Link/hooks.ts:40]: You can remove this part

File: components/Section/Section.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import waveWhiteBG from "./assets/waveWhite.png";
+import wavelight from "./assets/wave-light.png";
+
+export type BGColor =
+  | "wavelight"
+  | "grayGradient"
+  | "grayWave"
+  | "purple"
+  | "flatGray"
+  | "flatWhite";
```
Comment (iAdramelk) [inline components/Section/Section.tsx:12]: We can remove unused BGs here or maybe even fully remove section component.

**Review by iAdramelk — CHANGES_REQUESTED** [review]

File: components/Icon/icons.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+export { default as arrow } from "./svg/arrow.svg?react";
```
[the remark this answers is not in the document]
Comment (Alqanar) [reply components/Icon/icons.ts:1]: Yes, I had it set aside for later. Check which icons are used

File: components/Section/Section.tsx (hunk 2 above)
Comment (Alqanar) [reply components/Section/Section.tsx:12]: I deleted some that I'm sure are not used. I'll check the rest later, when all the necessary layouts and components are in the repository

File: components/Checkbox/Checkbox.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
+import { ChangeEvent, ReactNode, useCallback } from "react";
```
[the remark this answers is not in the document]
Comment (Alqanar) [reply components/Checkbox/Checkbox.tsx:1]: This component was used in Marketo, but I deleted that part

File: @types/window.d.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  interface Window {
+    MktoForms2?: {
+      loadForm: (
+        baseUrl: string,
+        munchkinId: string,
+        formId: string,
+        callback: () => void
+      ) => void;
+    };
+    teleport?: {
```
[the remark this answers is not in the document]
Comment (Alqanar) [reply @types/window.d.ts:11]: Yeah, I thought about that. But I thought I might have to redo it for the Blog.

[the remark this answers is not in the document]
Comment (iAdramelk) [reply @types/window.d.ts:11]: It is only used to integrate with the teleport login widget on `signup` page. No other pages use it.

[the remark this answers is not in the document]
Comment (Alqanar) [reply @types/window.d.ts:11]: Ah, I got it. Thanks

**Review by iAdramelk — APPROVED** [review]

**Pull request merged**